templates/master/01-master-kubelet/_base/units/kubelet.yaml

```diff
@@ -22,12 +22,12 @@ contents: |
         --kubeconfig=/var/lib/kubelet/kubeconfig \
         --container-runtime=remote \
         --container-runtime-endpoint=/var/run/crio/crio.sock \
-        --node-labels=node-role.kubernetes.io/master,node.openshift.io/os_id=${ID} \
+        --node-labels={{masterLabels .}} \
```

Comment:
One thing I discovered is that `--node-labels` docs clearly says:
> Labels to add when registering the node in the cluster

So changing it post-registration won't do anything.  Hence I think we shouldn't template this; the systemd unit should register with the labels/taints.  And then the MCO comes by later and updates them dynamically.

We also want to avoid using `MachineConfig` for this because e.g. changing a MC today involves rebooting and we clearly don't need to reboot for removing the master taint.

Reply:
This part was intended to make the NEXT node join correctly. Instead of getting 1 set of labels and then later having that set get updated. For existing nodes I update those:
https://github.com/openshift/machine-config-operator/pull/845/files#diff-a44cd14a0381ca49bcbaaf967b3cc9b0R556

The fact that an MC change requires a reboot is kinda stinky....

Reply:
> This part was intended to make the NEXT node join correctly.

Doesn't hurt to have a master start !schedulable briefly.  For workers...we may indeed need support for adding user-specified custom labels on bringup for special hardware and the like, but I think most cases of that end up adding taints instead post-registration.

Reply:
Adding labels and removing taints post-join is reasonable.  But adding a taint post join isn't really the best in my mind, since an inappropriate workload may have landed before the taint was added. So I sort of feel like the right taints should be added by the kubelet at join.

We could add every node with some megataint in the template. But then removing that taint would be required for every new pool. Which of course we can get right for our defaults, but is a likely bad stumbling block for users who aren't super savy.

I guess that means I feel like I can drop this part for labels, but the complete story seems weak if I drop it for taints....

Reply:
>  But adding a taint post join isn't really the best in my mind, since an inappropriate workload may have landed before the taint was added. 

True.